### PR TITLE
[stable/minio] Fix GCS credentials issue

### LINKED ISSUE #5911: [stable/minio] GCS Gateway Credentials not passed correctly
**Is this a request for help?**: No

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Bug Report

**Version of Helm and Kubernetes**: Helm 2.9.1, Kubernetes 1.10.2-gke.3 (running on GKE)

**Which chart**: stable/minio

**What happened**: Attempted to deploy Minio chart using GCS as backend gateway, credentials are not passed. Pods exited with error:
`Error: Unable to initialize gateway layer`

**What you expected to happen**:
Minio pod should start with GCS bucket as storage backend

**How to reproduce it** (as minimally and precisely as possible):
1. Download GCS service account credentials to JSON file (eg `our-project-credentials.json`)
2. Download default `values.yaml` file and apply the following changes:
```yaml
# values.yaml
persistence:
  enabled: false

gcsgateway:
  enabled: true
  # credential json file of service account key
  gcsKeyJson: "./our-project-credentials.json"
  # Google cloud project-id
  projectId: "our-project-123456"
```
3. Deploy helm chart:
```bash
helm install --name minio-production -f values.yaml stable/minio
```

```bash
$ kubectl get pods
NAME                                                    READY     STATUS    RESTARTS   AGE
minio-production-8498712-qjf99                        0/1       Error     0          11s

$ kubectl logs minio-production-8498712-qjf99
Created minio configuration file successfully at /root/.minio
Error: Unable to initialize gateway layer
```

**Anything else we need to know**:
* The README file is improperly formatted (last row in table is broken)
* Our GCS credentials work perfectly via a standard Kubernetes deployment of minio (we have rolled back to using this for now)
* When viewing the secret, it looks like it's only storing the JSON filename, not the contents of it
* It isn't clear exactly how to pass the JSON file (or its contents) to the Helm chart (via values or a helm `--set` argument), I have been unable to find a similar example to get this working

At a guess, I'm wondering if the `secrets.yaml` file shouldn't be this:
```yaml
{{- if .Values.gcsgateway.enabled }}
  gcs_key.json: {{ .Values.gcsgateway.gcsKeyJson | b64enc }}
{{- end }}
```
but instead something like this:
```yaml
{{- if .Values.gcsgateway.enabled }}
  gcs_key.json: {{ .Files.Get .Values.gcsgateway.gcsKeyJson | b64enc }}
{{- end }}
```

I spent longer than I would have liked trying to get this working, I assumed it was my own lack of understanding but now I'm thinking it might be that this was merged without testing?

Thanks

### PR DESCRIPTION
**Pull request by aelbarkani**
#### What this PR does / why we need it:
This PR fixes issue #5911 since PR #9477 seems to be on standby.
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #5911 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped

### CODE REVIEW

File: stable/minio/templates/deployment.yaml [hunk 1]
```diff
@@ -179,4 +176,4 @@ spec:
             - key: {{ .Values.tls.publicCrt }}
               path: CAs/public.crt
         {{ end }}
-{{- end }}
+{{- end }}
```
Comment (scottrigby) [inline stable/minio/templates/deployment.yaml]: Needs newline at EOF

**Review by scottrigby — COMMENTED** [review]
This is a duplicate of #9705 (minus the formatting fixes), right? Afaik that wasn't on standby just was not yet reviewed. Butler me know if I'm not reading that right, I'm looking at this on my phone omw to kubecon

File: stable/minio/templates/deployment.yaml (hunk 1 above)
Comment (aelbarkani) [reply stable/minio/templates/deployment.yaml]: solved

**Comment by aelbarkani** [comment]
Yes it was a duplicate of #9477 and #9705. Closing this PR.

**Pull request closed without merging**